site/profile/manifests/userportal.pp

```diff
@@ -0,0 +1,121 @@
+class profile::userportal::server (
+  $password
+){
+  $domain_name = lookup('profile::freeipa::base::domain_name')
+  package {['python3-virtualenv', 'python38', 'python38-devel']: }
```

Comment:
The equivalent virtualenv is packed with python3 since Python 3.5 in the module `venv`. The package python3-virtualenv is not necessary is this case.